### DISCUSSION TITLE
500: Internal Server Error with Next Js in some routes

### DISCUSSION THREAD

**Discussion by hexiaochen666** [vercel/next.js#38756] [Help]
In my local development, everything ran smoothly. However, this is an error I get when I run on Vercel, so I would like to get some suggestions on how to fix it.
![image](https://user-images.githubusercontent.com/52769073/179501195-f9a91bd9-c4f3-4bca-b75b-3b453ad88ec2.png)
I use getServerSideProps.
```
export async function getServerSideProps() {
  const res = await fetch('api')
  const {result} = await res.json()
  return {
    props:{
      data:result,
    }
  }
}
```

**Comment by icyJoseph** [comment]
Hi,

Without a repo or logs from Vercel this could be a whole lot of things.

- Attempting to get data from `api` routes from within `GSSP`
- Maximum payload reached
- Data in production has different shapes that those used in development
- etc, etc...

Could you try to share a repository link, or logs from Vercel?

**Reply by hexiaochen666** [reply]
export async function getServerSideProps() {
  const res = await fetch('api')
  const {result} = await res.json()
  return {
    props:{
      data:result,
    }
  }
}

**Reply by icyJoseph** [reply]
What's `'api'` in this context?

Regardless, that code, as shown, should fail because fetch, is actually polyfilled with node-fetch and it expects absolut URLs.

Using GetServerSideProps to reach for API routes is unnecessary, you are already on the server.

The next docs say it best, https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props#getserversideprops-or-api-routes:

> It can be tempting to reach for an API Route when you want to fetch data from the server, then call that API route from getServerSideProps. This is an unnecessary and inefficient approach, as it will cause an extra request to be made due to both getServerSideProps and API Routes running on the server.

**Reply by vixalien** [reply]
if anyone has this same problem again, I'll just share the experience. It's hard to debug this issue because when it happens, there are no logs in the Vercel environment whatsoever. My error iwas that I was using the `Event` class which is not available on Node.JS v14 (which my Vercel project was configured to use). So, I recommend you use a tool such as `nvm` to run your project with the same node version as the one on Vercel, and change your node version to the latest in the Vercel project's settings and redeploy.

**Reply by ttsoares** [reply]
How to know what Node version runs in Vercel ?

**Comment by Mahmoudgalalz** [comment]
on my side, I was doing a prisma data-proxy which caused that error, and I backed to the prisma generator and the issue was fixed 
so do not use data proxy just the mysql connection string and it will work

**Comment by Hakeemmidan** [comment]
In my case, I was getting the error on an endpoint that didn't have a response. Adding a `NextResponse` fixed it.

`next@13.4.4`

```js
export async function POST(request: Request) {
  try {
    // ... 
    return NextResponse.json({ hello: 'world' }); // <-- Needed to add a response, didn't have before
  } catch (error) {
    return NextResponse.json({ error: `Failed because of ${error}` }, { status: 500 });  
  }
```
